modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/EditFolderMVCActionCommand.java

```diff
@@ -232,9 +364,12 @@ private void _updateFolder(ActionRequest actionRequest)
 			long parentFolderId = ParamUtil.getLong(
 				actionRequest, "parentFolderId");
 
-			_dlAppService.addFolder(
+			Folder folder = _dlAppService.addFolder(
 				null, repositoryId, parentFolderId, name, description,
 				serviceContext);
+
+			_inheritPermissionFromParentFolder(
+				actionRequest, folder, serviceContext);
```

Comment:
Should move this change to DLFolderLocalServiceImpl as it will cause unexpected behavior if we add the folder via the API.